tests/utils/test_species_rarity.py

```diff
@@ -0,0 +1,49 @@
+import unittest
+from unittest.mock import Mock, patch, MagicMock
+
+from myning.config import SPECIES
+from myning.objects.character import CharacterSpecies
+from myning.objects.player import Player
+from myning.utils.species_rarity import get_current_tier, get_recruit_species
+
+class TestSpeciesRarity(unittest.TestCase):
+    """Test for Species Rarity"""
+    
+    @patch("myning.objects.player.FileManager.load", return_value=None)
+    @patch("builtins.input", return_value="TestPlayerName")
+    def test_initialize(self, mock_input, mock_load):
+        # Mock FileManager.load to return None, simulating a new player
+        mock_load.return_value = None
+
+        # Create a mock Character instance to represent the player
+        mock_player = MagicMock(spec=Player)
+        mock_player._allies = []
+        mock_player._fallen_allies = []
+        mock_player._discovered_species = []
+        
+        Player._instance = mock_player
+        Player.initialize()
+
+        return mock_player
+
+
+    def test_get_current_tier(self):
+        mock_player = self.test_initialize()
+
+        self.assertEqual(mock_player.discovered_species, [SPECIES[CharacterSpecies.HUMAN.value]])
+        self.assertEqual(get_current_tier(), 1)
```

Comment:
```suggestion
        assert get_current_tier() == 1
```
I prefer this style of assertions but I know we don't have established testing standards in this project. @TheRedPanda17 what do you think?

Reply:
I prefer `assert` as well.

Reply:
I have a couple more suggestions:

1. remove the class, this test can be a top-level function
2. test_initialize is a fixture, so rename it to `mock_player` and decorate it with @pytest.fixture
3. rename mock_player to player
4. instead of return mock_player, yield player
5. the fixture can be used like so:
```python
def test_get_current_tier(self, mock_player):
```

@mmorrison I'm happy to help with making these tests more pythonic!